packages/teams-js/src/internal/marketplaceUtils.ts

```diff
@@ -0,0 +1,59 @@
+import { marketplace } from '../public';
+
+/**
+ * @hidden
+ * Validates the cart item properties are valid
+ * @param cartItems The cart items
+ * @returns [true, undefined] if all properties of all cart items are valid, [false, error message] otherwise
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export function validateCartItems(cartItems: marketplace.CartItem[]): [boolean, string | undefined] {
+  for (const cartItem of cartItems) {
+    const priceValidationResult = validatePrice(cartItem.price);
+    if (!priceValidationResult[0]) {
+      return priceValidationResult;
+    }
+    const quantityValidationResult = validateQuantity(cartItem.quantity);
+    if (!quantityValidationResult[0]) {
+      return quantityValidationResult;
+    }
+  }
+  return [true, undefined];
```

Comment:
Rather than have this return [boolean, string | undefined], it would be simpler to just have it return only a boolean, and throw if it ever detects an invalid state. That way, if other functions use this in the future they will have the same error handling behavior